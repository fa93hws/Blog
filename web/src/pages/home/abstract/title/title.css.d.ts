// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'colorBlue': string;
  'title': string;
  'underline': string;
}
export const cssExports: CssExports;
export default cssExports;
