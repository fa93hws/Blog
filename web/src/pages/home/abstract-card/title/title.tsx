import * as React from 'react';
import * as classnames from 'classnames';
import { Link } from 'react-router-dom';
import { Title } from 'components/typography/typography';
import styles from './title.css';

type TitleProps = {
  children: React.ReactNode;
  linkTo: string;
  _forceHover?: boolean;
};
export const AbstractTitle = React.memo(
  ({ children, linkTo, _forceHover = false }: TitleProps) => (
    <div
      className={classnames(styles.title, {
        [styles.forceHover]: _forceHover,
      })}
    >
      <Link to={linkTo}>
        <Title.Medium>{children}</Title.Medium>
      </Link>
      <div className={styles.underline} />
    </div>
  ),
);
