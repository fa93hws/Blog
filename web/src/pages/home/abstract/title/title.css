@value colorBlue from '~components/colors.css';

.title {
  display: inline-block;
  color: colorBlue;
}

.underline {
  box-sizing: border-box;
  height: 2px;
  background-color: colorBlue;
  transform-origin: 50%;
  transform: scaleX(0%);
  transition: transform 300ms ease-in-out;
}

.title:hover > .underline {
  transform: scaleX(100%);
}
