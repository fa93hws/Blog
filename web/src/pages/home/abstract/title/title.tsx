import * as React from 'react';
import { Link } from 'react-router-dom';
import { Title } from 'components/typography/typography';
import styles from './title.css';

export const AbstractTitle = React.memo(
  ({ children, linkTo }: { children: React.ReactNode; linkTo: string }) => (
    <div className={styles.title}>
      <Link to={linkTo}>
        <Title.Small>{children}</Title.Small>
      </Link>
      <div className={styles.underline} />
    </div>
  ),
);
