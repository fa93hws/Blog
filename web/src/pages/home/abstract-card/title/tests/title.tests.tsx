import * as React from 'react';
import { MemoryRouter } from 'react-router';
import { AbstractTitle } from '../title';

describe('AbstractTitle', () => {
  it('renders the title for the abstract', () => {
    expect(
      <MemoryRouter>
        <AbstractTitle linkTo="target">Abstract Title</AbstractTitle>
      </MemoryRouter>,
    ).toMatchRenderedSnapshot();
  });
});
