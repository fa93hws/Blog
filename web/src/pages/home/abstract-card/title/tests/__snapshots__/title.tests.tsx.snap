// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbstractTitle renders the title for the abstract 1`] = `
<div
  class="title"
>
  <a
    href="/target"
  >
    <h3
      class="titleMedium"
    >
      Abstract Title
    </h3>
  </a>
  <div
    class="underline"
  />
</div>
`;
